### ISSUE TITLE
We strongly recommend passing in an `attention_mask` since your input_ids may be padded

### ISSUE THREAD

**Issue by hankyuwon** [hankyuwon/Emotional-Chatbot#4]
챗봇 테스트시 가끔 아래와 같은 에러 발생

```
python model.py --checkpoint model_EQ2A_Custom_Data_60label2map_120

user > 나도 노는 걸 좋아하는데 무시당하는 기분이야. 외로워
We strongly recommend passing in an `attention_mask` since your input_ids may be padded. See https://huggingface.co/docs/transformers/troubleshooting#incorrect-output-when-padding-tokens-arent-masked.
Chatbot > 혹시 주변에 이 문제에 대해 같이 고민할만하신다면 누구와 공감하며 전처럼 지낼 수 있을까<pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad> 진로에 대해 들한테 맞는 다양한 이들의 고민을 들으며 여유를 가지면서 이 상황을 잘 이해해줄 수 있는 주변 누군가가 생기길 진로에 응원하시길 응원하시길 응원하시니 걱정인가요
```

**Comment by hankyuwon** [comment]
답변 생성 부분에서 <pad>가 생성되는 지점을 파악

```
if args.checkpoint in 'noEmotion':
                    input_ids = torch.LongTensor(koGPT2_TOKENIZER.encode(Q_TKN + q + SENT + A_TKN + a)).unsqueeze(dim=0)
                else:
                    input_ids = torch.LongTensor(koGPT2_TOKENIZER.encode(emo_token + SENT + Q_TKN + q + SENT + A_TKN + a)).unsqueeze(dim=0)

                print('input size = ', len(input_ids[0]))
                print('input shape =', input_ids.shape)
```

```
input size =  39
input shape = torch.Size([1, 39])
q =  다음번 흡연하러 나가기 전에 담배를 들고 나가라고 이야기해야 할 것 같아.
a =   친구와 대화로 잘 풀어서 또 흡연할 때 옆집 할머니와 친해지
generative =  길


input size =  40
input shape = torch.Size([1, 40])
q =  다음번 흡연하러 나가기 전에 담배를 들고 나가라고 이야기해야 할 것 같아.
a =   친구와 대화로 잘 풀어서 또 흡연할 때 옆집 할머니와 친해지길
generative =  <pad>


input size =  41
input shape = torch.Size([1, 41])
q =  다음번 흡연하러 나가기 전에 담배를 들고 나가라고 이야기해야 할 것 같아.
a =   친구와 대화로 잘 풀어서 또 흡연할 때 옆집 할머니와 친해지길<pad>
generative =  <pad>
```

**Comment by hankyuwon** [comment]
대부분의 경우 패딩 토큰을 무시하기 위해 attention_mask를 제공해야 한다.
```
attention_mask = torch.tensor([[1, 1, 1, 1, 1, 1], [1, 0, 0, 0, 0, 0]])
output = model(input_ids, attention_mask=attention_mask)
print(output.logits)
>>> tensor([[ 0.0082, -0.2307], [-0.1008, -0.4061]], grad_fn=<AddmmBackward0>)
```

참고 : https://huggingface.co/docs/transformers/troubleshooting#incorrect-output-when-padding-tokens-arent-masked

**Comment by hankyuwon** [comment]
GPT2_INPUTS_DOCSTRING
```
**attention_mask**: (`optional`) ``torch.Tensor`` of shape ``(batch_size, sequence_length)``:
            Mask to avoid performing attention on padding token indices.
            Mask values selected in ``[0, 1]``:
            ``1`` for tokens that are NOT MASKED, ``0`` for MASKED tokens.
```

GPT2Attention 코드
```
if attention_mask is not None:
            # Apply the attention mask
            attn_weights = attn_weights + attention_mask
```

**Comment by hankyuwon** [comment]
> 답변 생성 부분에서 `<pad>` 가 생성되는 지점을 파악
> 
> ```python
> # test code
> while 1:
>    q = input("user > ").strip()
>    if q == "quit":
>       break
>    a = ""
>    emo_token, emo_str = Q2E_model.predict(q)
>    # print('Predicted Emotion is ',emo_str)
>             
>    while 1:
>       if args.checkpoint in 'noEmotion':
>          input_ids = torch.LongTensor(koGPT2_TOKENIZER.encode(Q_TKN + q + SENT + A_TKN + a)).unsqueeze(dim=0)
>       else:
>          input_ids = torch.LongTensor(koGPT2_TOKENIZER.encode(emo_token + SENT + Q_TKN + q + SENT + A_TKN + a)).unsqueeze(dim=0)
> 
>       pred = model(input_ids.to(device))
>       pred = pred.logits
>       gen = koGPT2_TOKENIZER.convert_ids_to_tokens(torch.argmax(pred, dim=-1).squeeze().cpu().numpy().tolist())[-1]
>       if gen == EOS:
>          break
>       a += gen.replace("▁", " ")
> 
> 
>       # Debug
>       print('input size = ', len(input_ids[0]))
>       print('input shape =', input_ids.shape)
>       print('q = ', q)
>       print('a = ', a)
>       print('generative = ', gen)
>       
>    print("Chatbot > {}".format(a.strip()))
> ```
> 
> ```
> input size =  39
> input shape = torch.Size([1, 39])
> q =  다음번 흡연하러 나가기 전에 담배를 들고 나가라고 이야기해야 할 것 같아.
> a =   친구와 대화로 잘 풀어서 또 흡연할 때 옆집 할머니와 친해지
> generative =  길
> 
> 
> input size =  40
> input shape = torch.Size([1, 40])
> q =  다음번 흡연하러 나가기 전에 담배를 들고 나가라고 이야기해야 할 것 같아.
> a =   친구와 대화로 잘 풀어서 또 흡연할 때 옆집 할머니와 친해지길
> generative =  <pad>
> 
> 
> input size =  41
> input shape = torch.Size([1, 41])
> q =  다음번 흡연하러 나가기 전에 담배를 들고 나가라고 이야기해야 할 것 같아.
> a =   친구와 대화로 잘 풀어서 또 흡연할 때 옆집 할머니와 친해지길<pad>
> generative =  <pad>
> ```

우선, model의 input size가 초기 설정한 최대 시퀀스의 길이(max_len)을 넘어서는 지점에서 에러가 발생하기 때문에, 최대 길이를 넘어서지 않게 코드 수정
